src/plugin_discovery/__tests__/find_plugin_specs.js

```diff
@@ -0,0 +1,72 @@
+import { resolve } from 'path';
+import { readdirSync } from 'fs';
+
+import expect from 'expect.js';
+import { findPluginSpecs } from '../find_plugin_specs';
+import { PluginSpec } from '../plugin_spec';
+
+const CORE_PLUGINS = resolve(__dirname, '../../core_plugins');
+
+describe('plugin discovery', () => {
+  describe('findPluginSpecs()', function () {
+    this.timeout(10000);
+
+    it('finds specs for specified plugin paths', async () => {
+      const { spec$ } = findPluginSpecs({
+        plugins: {
+          paths: [
+            resolve(CORE_PLUGINS, 'console'),
+            resolve(CORE_PLUGINS, 'elasticsearch'),
```

Comment:
Relying on arbitrary core plugins for this test doesn't seem like a good option.  It couples this nice new standalone module to two unrelated modules.  I would never in a million years expect the complete removal of renaming of `console` or `elasticsearch` to break the tests in this module.